rustls/src/client/hs.rs

```diff
@@ -219,10 +219,12 @@ fn emit_client_hello_for_retry(
         supported_versions.push(ProtocolVersion::TLSv1_2);
     }
 
-    let mut exts = Vec::new();
-    if !supported_versions.is_empty() {
-        exts.push(ClientExtension::SupportedVersions(supported_versions));
+    if supported_versions.is_empty() {
```

Comment:
I believe this is statically unreachable, https://github.com/rustls/rustls/blob/main/rustls/src/builder.rs#L224.

Reply:
In that case, I guess we should just `assert!(!supported_versoins.is_empty());` here.

Reply:
Or see #913.